organization_settings: Convert name to pills.

sujalshah-bit opened the pull request:
Converting plain links to pills for user and bot names across the User groups, Users, and Deactivated users panels. The conversion includes both bot names and their respective owner names. used same `user_display_only_pill.hbs` template to convert name to pills in the organization settings 

- Updated `user_display_only_pill.hbs` to utilize the newly created user_pill.hbs partial for displaying user and bot names as pills.
- Implemented the `user_pill.hbs` partial to encapsulate the common structure and styling of the pills, reducing code repetition.
- Added tooltip so that long names can seen from tooltip

Fixes #24229.

## Users

| Dark | Light |
|--------|-------|
| ![image](https://github.com/zulip/zulip/assets/73663475/0ed2103a-24c9-4d75-b3ee-ffef312531dd) | ![image](https://github.com/zulip/zulip/assets/73663475/4ad7cf3b-7672-43f2-852f-a8ab773cf3b6) |

## Bots
| Dark | Light |
|--------|-------|
| ![image](https://github.com/zulip/zulip/assets/73663475/f2e84a5b-06ee-4cbb-8983-011166a3cc7a) | ![image](https://github.com/zulip/zulip/assets/73663475/363caf04-e312-466d-81a6-3344f8daba77) |

## Deactivated users

| Dark | Light |
|--------|-------|
| ![image](https://github.com/zulip/zulip/assets/73663475/6634b6bf-9d07-4f4c-87e3-4b9131c68cd8) | ![image](https://github.com/zulip/zulip/assets/73663475/72d4bc09-bf4d-452d-9660-bd24013e043e) |

## tooltip
| tooltip | 
|--------|
| ![image](https://github.com/zulip/zulip/assets/73663475/199a17c3-ec4d-4198-9ab9-293ddca3e73f) |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
@sujalshah-bit Please add screenshots of your changes and complete the self-review checklist.

sujalshah-bit commented:
@alya  updated the screenshot

sahil839 commented on web/src/tippyjs.ts:
```diff
@@ -426,6 +426,21 @@ export function initialize(): void {
+        onHidden(instance) {
+            instance.destroy();
+        },
+    });
```
Can't we directly add `data-tippy-content` attribute to the html element instead of doing this?

sujalshah-bit commented on web/styles/right_sidebar.css:
```diff
@@ -214,6 +214,7 @@ $user_status_emoji_width: 24px;
 
 .my_user_status {
     opacity: 0.5;
+    white-space: nowrap;
```
`white-space: nowrap;` prop is added so that "(you)" in pill don't break. not for right sidebar


![image](https://github.com/zulip/zulip/assets/73663475/c3ccda36-8825-41f4-a373-defc886e9474)

sahil839 replied:
Yeah, I just wanted to make sure that this does not break the tooltip in right sidebar because same class is used there as well.

sujalshah-bit commented on web/templates/settings/admin_user_list.hbs:
```diff
@@ -1,13 +1,6 @@
-        {{else}}
-        <i class="fa fa-ban deactivated-user-icon tippy-zulip-delayed-tooltip" data-tippy-content="{{t 'User is deactivated' }}" {{#if is_active}}style="display: none;"{{/if}}></i>
-        {{/if}}
+    <td class="panel_subscriber_member_list">
```
now using `panel_member_list`.

sujalshah-bit commented:
@sahil839  made the changes kindly review :)

sahil839 commented on web/src/tippyjs.ts:
```diff
@@ -426,6 +426,24 @@ export function initialize(): void {
+                {
+                    name: "flip",
+                    options: {
+                        fallbackPlacements: "bottom",
```
This code will not be needed if you add `tippy-zulip-delayed-tooltip` class to html element.

sahil839 reviewed (commented):
Posted a couple of comments.

sujalshah-bit commented:
@sahil839  made the changes kindly review:)
- added a separate prep commit for `panel_user_list`
- added 'tippy-zulip-delayed-tooltip' to html

I have check it is not breaking in right sidebar
![image](https://github.com/zulip/zulip/assets/73663475/ece44af9-b9c5-48ef-b09f-257bcea74eb8)

sahil839 reviewed (commented):
Posted a few comments. I missed some of the things in previous reviews, so sorry for that.

The commit message for the first prep commit should be improved - there could be a better prefix than `prep_commit` like `templates` or `settings` and you should mention in the commit message that why the change is being done.

alya commented on web/templates/user_pill.hbs:
```diff
@@ -0,0 +1,15 @@
+        <img class="pill-image" src="{{img_src}}" />
+    {{/if}}
+    <span class="pill-value">{{display_value}}</span>
+    {{#unless is_active}}
```
I guess the ideal design would be to use the slashed circle on top of the avatar, as in the pills illustration in #26833, though I think we don't want to fade the pill in this context.

Barring that, I guess just the slashed circle as in the screenshot in this PR seems fine for now.

alya commented:
@sujalshah-bit What's the status of this PR? Has @sahil839 's feedback above been addressed?

sujalshah-bit commented on web/templates/user_display_only_pill.hbs:
```diff
@@ -1,14 +1,9 @@
+            {{> user_pill user_id=bot_owner_id}}
+        {{/unless}}
+    {{else}}
+        {{> user_pill user_id=user_id}}
```
I didn't understand your point; it is quite confusing. I will explain why I chose this approach: in the organization setting, there are cases where bots are present, and their owners are also displayed. So, for that purpose, when we want to display the owner, I use `is_bot` as false, and when we want to display bots, I use `is_bot` as true. I don't see where we are passing incorrect values.

sahil839 replied:
The first instance of `user_display_only_pill` will be called for bots when showing the "Bots" list in organization settings, is that correct? And you have set the `is_bot` field to `false` for it.

sujalshah-bit replied:
made the changes. :) sorry for inconvenience

sujalshah-bit commented on web/templates/stream_settings/new_stream_user.hbs:
```diff
@@ -1,6 +1,6 @@
-    <td class="panel_subscriber_member_list">
-        {{> ../user_display_only_pill display_value=full_name}}
+    <td class="panel_user_list">
+        {{> ../user_display_only_pill display_value=full_name is_active=true}}
```
done I have added these changes in next separate commit.

sahil839 replied:
This should be in the commit which adds `is_active` field to `user_display_only_pill`, so you can add this in the "Convert name to pills" commit.

sujalshah-bit commented:
@sahil839  added `is_active` field changes in separate commit and posted comment above kindly reply. sorry for the late response

sujalshah-bit commented:
made the changes @sahil839

sahil839 commented on web/templates/user_pill.hbs:
```diff
@@ -0,0 +1,15 @@
+<a data-user-id="{{user_id}}" class="view_user_profile pill" tabindex="0">
```
We do not use this template anywhere else currently except in `user_display_only_pill`. Is this extracted out considering some future use-case.

sujalshah-bit replied:
`user_pill.hbs` is removed the main purpose of that file was to use the same code in if and else block of `is_bot` section. However the `is_bot` is removed so this file is also not needed.

Posted a comment below kindly reply.

sujalshah-bit commented:
@sahil839, I'd like to address something. While playing with PRs, I found a bug in my code. When a bot is deactivated, the owner of that bot also gets the deactivated icon. This occurs because when the bot is deactivated, `is_active` is set to false, which is then automatically passed on to `user_display_only_pill.hbs`.

![image](https://github.com/zulip/zulip/assets/73663475/99343bf4-5974-418f-88b2-55032bda1aff)

```hbs
{{#if no_owner }}
    {{bot_owner_full_name}}
{{else}}
    {{> ../user_display_only_pill display_value=bot_owner_full_name user_id=bot_owner_id img_src=owner_img_src}}
    {{!-- Here, is_active is passed as false if the bot is deactivated. --}}
{{/if}}
```

To address this, I've made a change by passing `is_active` as true:

```hbs
{{> ../user_display_only_pill display_value=bot_owner_full_name user_id=bot_owner_id img_src=owner_img_src  is_active=true}}
```
![image](https://github.com/zulip/zulip/assets/73663475/e248f96e-1493-4e20-acb7-ab6ba8d64f3e)


However, consider a scenario where if a user is deactivated, all the bots created by that user are also deactivated. In this case, we should show the deactivated icon in the user pill as well, particularly in the bots section. Although I've made changes to match our current behavior, which is to not show the deactivated icon in the bots section's user pills if the user is deactivated.

**Current behaivor: ( Here user-26 is deactivated ) but does not show deactivated icon**
![image](https://github.com/zulip/zulip/assets/73663475/55eae794-a319-499e-845f-f6dca3807297)

**If user is deactivated the icon should also be shown in the pill in Bot section:**
![image](https://github.com/zulip/zulip/assets/73663475/3c8b9d06-0316-4a76-b809-d5659e256170)

sahil839 commented:
Yes, I guess we can show the deactivated icon in user pills if the user (i.e. bot owner) is deactivated.

alya commented:
> Yes, I guess we can show the deactivated icon in user pills if the user (i.e. bot owner) is deactivated.

Sounds good to me!

timabbott commented:
@sujalshah-bit let me know when this is ready for further review.

sujalshah-bit commented:
Closing in favor of #30371
